OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/alias/AllocationSiteBasedAliasSet.scala

```diff
@@ -0,0 +1,32 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj.tac.fpcf.analyses.alias
+
+import scala.collection.mutable
+
+import org.opalj.br.PC
+import org.opalj.br.fpcf.properties.Context
+
+class AllocationSiteBasedAliasSet extends AliasSetLike[AllocationSite, AllocationSiteBasedAliasSet] {
+
+    private var _pointsTo: mutable.Set[AllocationSite] = mutable.Set.empty[AllocationSite]
+
+    def addPointsTo(context: Context, pc: PC): Unit = {
+        addPointsTo((context, pc))
+    }
+
+    def pointsTo(context: Context, pc: PC): Boolean = _pointsToAny || pointsTo((context, pc))
+
+    def allPointsTo: mutable.Set[AllocationSite] = _pointsTo
+
+    def intersection(other: AllocationSiteBasedAliasSet): AllocationSiteBasedAliasSet = {
+        val intersection = new AllocationSiteBasedAliasSet
+
+        // we could theoretically optimize this by stopping once two intersecting elements have been found
```

Comment:
Then why don't you do that? You could just change the method to return, e.g., `(Boolean, Boolean)` and replace the intersection with something like `_pointsTo.exists(other._pointsTo.contains)`.